miden-lib/asm/sat/account.masm

```diff
@@ -66,3 +84,46 @@ end
 export.get_initial_hash
     exec.layout::get_init_acct_hash
 end
+
+#! Returns the account type of the account the transaction is being executed against.
+#!
+#! The account type can be of the following forms:
+#! - regular account with updatable code: REGULAR_ACCOUNT
+#! - regular account with immutable code: NON_UPDATABLE_ACCOUNT
+#! - fungible asset faucet account with immutable code: FUNGIBLE_FAUCET_ACCOUNT
+#! - non-fungible asset faucet account with immutable code: NON_FUNGIBLE_FAUCET_ACCOUNT
+#!
+#! Stack: []
+#! Output: [acct_type]
+#!
+#! - acct_type is the account type.
+export.type
+    # get the account id
+    exec.layout::get_acct_id
+    # => [acct_id]
+
+    # compute the account type
+    u32split swap drop push.30 u32checked_shr
```

Comment:
We should use `u32checked_shr.30` here it is way more efficient than the one which reads the number of bits to shift by from the stack (i.e., 4 vs. 47 cycles).